intake/source/base.py

```diff
@@ -215,6 +222,27 @@ def to_spark(self):
         """
         raise NotImplementedError
 
+    @property
+    def entry(self):
+        if self._entry is None:
+            raise ValueError("Source was not made by a catalog")
+        if self._entry() is None:
+            raise RuntimeError("Tried to recover entry, but it was gone")
+        return self._entry()
+
+    def clone(self, **kwargs):
+        return self.entry(**kwargs)
+
+    def __call__(self, *args, **kwargs):
+        return self.clone(**kwargs)
+
+    def describe(self):
+        return self.entry.describe()
+
+    @property
```

Comment:
This seems like an application layer thing that should not be baked into the core objects.  If this is something that was there and needs to stay for back compatibility can we deprecate it?

Reply:
Right, you could construct a GUI for an entry previously (which allowed you to select the user parameters and then plot with point&click); so if the user is no longer exposed to the entry object, this is the only way to do it. Of course, we could question the usefulness of that particular GUI view, but otherwise the source no longer has any connection to the GUIs at all.

Reply:
It makes a lot more sense to me to pass the `DateSrouce` to the gui (`make_the_clicky_gui(ds)`) rather than have it be a method so that it is possible to support exposing several gui generation tactics to the user (ex "all the details" vs "executive summary").

Reply:
I too would like to revisit this, but I think should not lump that into this PR.

Reply:
The GUI code can probably do with a refactor following improvements in https://github.com/intake/filesystem_spec/blob/master/fsspec/gui.py#L14